ovc/disks.go

```diff
@@ -85,6 +84,85 @@ type DiskList []struct {
 	AccountID   int         `json:"accountId"`
 }
 
+// DiskExposeProtocolNBD is a constant representing the storage protocol NBD
+// (Network Block Device)
+const DiskExposeProtocolNBD = "nbd"
+
+// DiskExposeConfig is used to request that a given disk is exposed via a given
+// cloudspace and protocol
+type DiskExposeConfig struct {
+	Protocol     string `json:"-"`
+	DiskID       int    `json:"diskId"`
+	CloudSpaceID int    `json:"cloudspaceId"`
+}
+
+// DiskEndPointDescriptor is an interface that a type representing a storage
+// protocol endpoint shall implement to facilitate accessing protocol specific
+// information in DiskExposeInfo
+type DiskEndPointDescriptor interface {
+	Protocol() string
+}
+
+// NBDDiskEndPointDescriptor contains NBD specific information on how to
+// access a disk exposed as a Network Block Device
+type NBDDiskEndPointDescriptor struct {
+	Address string `json:"address"`
+	Port    int    `json:"port"`
+	Name    string `json:"name"`
+	User    string `json:"user"`
+	Psk     string `json:"psk"`
+}
+
+// Protocol implements DiskEndPointDescriptor.Protocol
+func (*NBDDiskEndPointDescriptor) Protocol() string {
+	return DiskExposeProtocolNBD
+}
+
+// DiskExposeInfo contains information on how to access a disk exposed using
+// a given storage protocol.
+type DiskExposeInfo struct {
+	Protocol string                 `json:"protocol"`
+	EndPoint DiskEndPointDescriptor `json:"endpoint"`
+}
+
+// UnmarshalJSON deserializes DiskInfo from a buffer containing a JSON
+// representation of DiskInfo
+func (i *DiskExposeInfo) UnmarshalJSON(b []byte) error {
+	type ProtoProbe struct {
+		Protocol string `json:"protocol"`
+	}
+
+	probe := ProtoProbe{}
```

Comment:
Since I don't immediately see ProtoProbe being used anywhere else you could use an anonymous struct:
```
probe := struct {
    Protocol string `json:"protocol"`
}{}
```
Just FYI if you don't know it, for me it can stay the way it is